lib/tbot/service_bot_identity.go

```diff
@@ -292,7 +299,10 @@ func (s *identityService) renew(
 	if s.cfg.Onboarding.RenewableJoinMethod() {
 		// When using a renewable join method, we use GenerateUserCerts to
 		// request a new certificate using our current identity.
-		authClient, err := clientForIdentity(ctx, s.log, s.cfg, currentIdentity, s.resolver)
+		// We explicitly create a new client here to ensure that the latest
```

Comment:
This is the one place where we actually need to ensure we have a client which is using the latest identity - so we explicitly create the client. This is because of the generation counter stored in the x509 cert which is compared with the one on the user during the certificate renewal process. If an older identity is still in use by a client, the bot will be locked.